### DISCUSSION TITLE
Prod deployment is showing 500 internal server error

### DISCUSSION THREAD

**Discussion by marekadvocate** [vercel/vercel#7895] [Help]
Hello, we are 2 days facing production deployment problems on next js application

We got this issues 

> 500 internal server error

> Serverles funcion has crashed

Everything seems be builded, deployed correctly, but nothing else is possible to get from logs,

Does anybody has idea what could be the problem ?

**Comment by marekadvocate** [comment]
@leerob

**Comment by marekadvocate** [comment]
<img width="648" alt="Snímka obrazovky 2022-05-31 o 14 06 26" src="https://user-images.githubusercontent.com/97805406/171169336-3baff5cc-15bb-47ce-bad3-c842b5b11063.png">

**Comment by 0xJohnnyboy** [comment]
I'm facing the same issue on a project. But it only happens on a specific page that makes an API call to Notion. I reached out to the support team

**Comment by mathiasabdissa** [comment]
I am facing the same issue too with Vercel Node.js Application

**Comment by amyegan** [comment]
The [Function Logs](https://vercel.com/docs/concepts/deployments/logs#function-logs) should reveal more about the error. These logs are generated on-demand, so make sure you have the log page open before visiting the failing page in another tab or window.

**Comment by mfyz** [comment]
Same issue, no errors displayed in the logs view. Logs view show my request, and the lambda start log line, then my custom console.log lines up until where it fails but then nothing in that particular request session. Basically I get zero error output in vercel dashboard logs view regardless of how long I keep it open.

Same as the log drains. I use sematext for forwarding my vercel logs. Sematext registers that those HTTP requests were responded with 500 and I do see same partial log output as the "errored" logs but only shows the successful lambda start marker, my console.logs up until to the point that my function crashes. But nothing regarding to the reason it crashed...

Note: This only happens on vercel and not on my local - The first instance of this I was able to debug line by line with dozen+ deployments were pointing issue to an area that I was doing a redirection using res.setHeader in getServerSideProps. I ended up responding body with <script>location.href... in order to get my redirection working. Not ideal but works. Now I'm detecting other requests to my users returning 500 without any clue. This is not an error I can reproduce, but I can monitor and observe my requests on sematext (basically hosted ELK stack).

**Reply by kdeblois** [reply]
Having exactly the same problem as m@marekadvocate. Did you ever resolve it? The functiion log suggests it has something to no with node.js. Everything works fine locally.
